gui/src/hooks/config.ts

```diff
@@ -73,9 +60,21 @@ export function useConfigProvider(): ConfigContext {
     loadConfig: async () => {
       setLoading(true);
       try {
-        const json = await readTextFile('config.json', {
-          dir: BaseDirectory.App,
-        });
+        const migrated = localStorage.getItem('configMigrated');
+        if (!migrated) {
+          const oldConfig = await readTextFile('config.json', {
+            dir: BaseDirectory.App,
+          }).catch(() => null);
+
+          if (oldConfig) localStorage.setItem('config.json', oldConfig);
+
+          localStorage.setItem('configMigrated', 'true');
+        }
+
+        const json = localStorage.getItem('config.json');
+
+        if (!json) throw new Error('Config has ceased existing for some reason');
```

Comment:
This error is a little strange, as you'd get it before any config is written, so it's not necessarily happening when the config vanishes

Reply:
it can also appear if you remove the config while its gonna get read :P. But it's an error that shouldn't happen at all unless user error or weird behaviour

Reply:
eh it's fine whatever